R/merge_sce_list.R

```diff
@@ -0,0 +1,200 @@
+#' Merge a list of SCEs as preparation for formal integration
+#'
+#' This function takes an optionally-named (if named, ideally by a form of
+#'  library ID) list of SingleCellExperiment (SCE) objects and merges them into
+#'  one SCE object. At least some genes must be present in all SCEs in order to
+#'  merge them. Currently any present altExps are not retained.
+#'
+#'  Original SCE contents are modified or retained as follows:
+#'  - The resulting colData slot will include a new column specified by
+#'    `batch_column` (default "library_id") that either holds the originating SCE
+#'    object's name (referred to as `sce_name` here), of if it is unnamed then
+#'    its index in the provided `sce_list`.
+#'  - The resulting colData slot will include another new column `cell_id_column`
+#'    (default "cell_id") that is formatted as `{sce_name}-{rowname}`, where
+#'    `sce_name` is the given batch (library) value and `rowname` is the colData
+#'    row's rowname which often, but not always, holds a unique cell barcode.
+#'    This column serves to match cells back to both originating batch and cell name.
```

Comment:
I would agree with Josh's comment that this should just be the original barcode or `rownames(sce)` prior to merging.